benchmark/spmv/cuda_linops.hpp

```diff
@@ -65,23 +65,24 @@ class CuspBase : public gko::LinOp {
              const gko::dim<2> &size = gko::dim<2>{})
         : gko::LinOp(exec, size)
     {
-        gpu_exec = dynamic_cast<const gko::CudaExecutor *>(exec.get());
-        if (gpu_exec == nullptr) {
+        gpu_exec_ = std::dynamic_pointer_cast<const gko::CudaExecutor>(exec);
+        if (gpu_exec_ == nullptr) {
             GKO_NOT_IMPLEMENTED;
         }
         this->initialize_descr();
     }
 
     CuspBase &operator=(const CuspBase &other)
     {
-        this->gpu_exec = other.get_gpu_exec();
+        this->gpu_exec_ = std::dynamic_pointer_cast<const gko::CudaExecutor>(
+            other.get_executor());
```

Comment:
Here, it could happen that if you have three `CuspBase` objects `a`, `b` and `c`, you will get a different executor with this change than before:
```
b = a;    // b.gpu_exec_ ^= a.get_executor()
c = b;    // c.gpu_exec_ ^= b.get_executor()  != a.get_executor()
```
with the previous code, you would have `c.gpu_exec_ ^= a.get_executor()`.
I think you actually want this behavior right now, but since I am not sure, I wanted to mention it.

Reply:
I nonetheless added a call to the `LinOp::operator=()` for good measure. This should take care of properly setting the executor.

Reply:
Actually, the LinOp `operator=` does not change the executor. `PolymorphicObject` is the class that takes care of the executor, and the `operator=` function looks like this:
https://github.com/ginkgo-project/ginkgo/blob/develop/include/ginkgo/core/base/polymorphic_object.hpp#L76
The comment above even explicitly says `preserve the executor of the object`, so the executor will not be set in a call for `=`.
I am not sure if you want the executor to change. If you do want the `this->gpu_exec_` to change to `other->gpu_exec_`, i think you can only do
```
this->gpu_exec_ = other->gpu_exec_;
```

Reply:
Well everything should be propagated to `PolymorphicObject`, though you are right I forgot we are preserving the executor.

I will go with allowing the `gpu_exec_` to change as in fact the current version is slightly incorrect: `GKO_NOT_IMPLEMENTED` would have to be thrown in some cases.